### DISCUSSION TITLE
How to user httxRunner.RunEnumeration() to store data so it can be printed as HTML?

### DISCUSSION THREAD

**Discussion by MarsIsRed** [projectdiscovery/httpx#974] [Q&A]
![image](https://user-images.githubusercontent.com/124643733/217913011-d9581e68-0228-4c6f-9ffc-d9afb272994f.png)

Error: httpxRunner.RunEnumeration() (no value) used as value

**Comment by olearycrew** [comment]
@Mzack9999  @ehsandeep can one of you take a look at this question? Thanks!

**Comment by aringo-bf** [comment]
If you look at the example script in [examples/example.go](https://github.com/projectdiscovery/httpx/blob/da012ccf25c0c5ad5979fbbaee26e9add4546166/examples/example.go) you can see the basics of how to use the runner. Almost all the PD tools operate by passing an option struct which lets you define what to scan for and how to store it.  HTTPX uses a callback in the option so the result can be processed further. 

The result struct is easy to work with - so in that callback just call any values in the [result struct ](https://github.com/projectdiscovery/httpx/blob/da012ccf25c0c5ad5979fbbaee26e9add4546166/runner/runner.go#L2010)  but be sure to handle if they errors and check for values. Then write them to an HTML template or store them in a JSON to be loaded by your HTML page.. etc. 

```
	options := runner.Options{
     
		OnResult: func(r runner.Result) {	
                      # call another function to write HTML or do it here using values from the r runner.Result
		},
	}
```
Hope this helps.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
